endpoint/ingress/ingress.go

```diff
@@ -0,0 +1,262 @@
+		if ingress.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			return true, nil
+		}
+		if ingress.Status.LoadBalancer.Ingress[0].IP != "" {
+			return true, nil
+		}
+	}
+	i.logger.Info("endpoint is unhealthy")
+	return false, nil
+}
+
+func (i *ingress) MarkForCleanup(ctx context.Context, c client.Client, key, value string) error {
+	i.logger.Info("marking endpoint evc for cleanup")
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.namespacedName.Name,
+			Namespace: i.namespacedName.Namespace,
+		},
+	}
+	err := utils.UpdateWithLabel(ctx, c, svc, key, value)
+	if err != nil {
+		i.logger.Error(err, "failed to mark endpoint svc for cleanup", "svc", i)
+		return err
+	}
+	ingress := &networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.namespacedName.Name,
+			Namespace: i.namespacedName.Namespace,
+		},
+	}
+	err = utils.UpdateWithLabel(ctx, c, ingress, key, value)
+	if err != nil {
+		i.logger.Error(err, "failed to mark endpoint ingress for cleanup", "ingress", i)
+		return err
+	}
+	return nil
+}
+
+// AddToScheme should be used as soon as scheme is created to add
+// core  objects for encoding/decoding
+func AddToScheme(scheme *runtime.Scheme) error {
+	return corev1.AddToScheme(scheme)
+}
+
+// APIsToWatch give a list of APIs to watch if using this package
+// to deploy the endpoint
+func APIsToWatch() ([]client.Object, error) {
+	return []client.Object{
+		&corev1.Service{},
+		&networkingv1.Ingress{}}, nil
+}
+
+// New creates an ingress endpoint object, deploys the resources on  the cluster
+// and then checks for the health of the loadbalancer. Before using the fields
+// it is always recommended to check if the loadbalancer is healthy.
+//
+// In order to generate the right RBAC, add the following lines to the Reconcile function annotations.
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch;delete
+func New(ctx context.Context, c client.Client, logger logr.Logger,
+	namespacedName types.NamespacedName,
+	ingressClassName *string,
+	subdomain *string,
+	labels, ingressAnnotations map[string]string,
+	ownerReferences []metav1.OwnerReference) (endpoint.Endpoint, error) {
+	ingressLogger := logger.WithValues("ingress", namespacedName)
+
+	ingressEndpoint := &ingress{
+		logger:             ingressLogger,
+		namespacedName:     namespacedName,
+		labels:             labels,
+		ingressAnnotations: ingressAnnotations,
+		ownerReferences:    ownerReferences,
+		backendPort:        backendPort,
+		ingressPort:        ingressPort,
+		ingressClassName:   ingressClassName,
+		subdomain:          subdomain,
+	}
+
+	if ingressClassName == nil {
+		ingressLogger.Info("ingress class not specified, using default ingress class in the cluster")
+	}
+
+	if subdomain == nil {
+		return nil, fmt.Errorf("subdomain cannot be nil")
+	} else {
```

Comment:
since the above, if block is returning, remove this else block and bring the code out?